addons/mail/models/mail_thread.py

```diff
@@ -1234,19 +1240,23 @@ def message_route(self, message, message_dict, model=None, thread_id=None, custo
         is_a_reply = bool(mail_messages)
 
         # 1.1 Handle forward to an alias with a different model: do not consider it as a reply
-        if reply_model and reply_thread_id:
-            other_alias = Alias.search([
-                '&',
+        if is_a_reply and reply_model and reply_thread_id:
+            other_model_aliases = Alias.search([
                 ('alias_name', '!=', False),
-                ('alias_name', '=', email_to_localpart)
+                ('alias_name', 'in', email_to_localparts),
+                ("alias_model_id.model", "!=", reply_model),
             ])
-            if other_alias and other_alias.alias_model_id.model != reply_model:
+            if other_model_aliases:
                 is_a_reply = False
+                rcpt_tos_valid_localparts = [to for to in rcpt_tos_valid_localparts if to in other_model_aliases.mapped('alias_name')]
```

Comment:
If I'm not wrong, this is calling `mapped()` in each loop, which seems an O<sup>n</sup> complexity.

What about this?

```suggestion
                rcpt_tos_valid_localparts = list(set(rcpt_tos_valid_localparts).intersection(other_model_aliases.mapped('alias_name')))
```